### ISSUE TITLE
Add `isCompressible()` method to fluid properties

### ISSUE THREAD

**Issue by joshuahansel** [idaholab/moose#24212]
## Reason
Some equations of state do not make sense to be used in certain flow models. For instance, incompressible equations of state do not make sense to be used with flow models where the density and specific internal energy are solved independently, like the Euler equations.

## Design
This would add a method `isCompressible()` to `SinglePhaseFluidProperties`. Then flow models can query this method if they need to.

## Impact
New virtual method in `SinglePhaseFluidProperties`.

**Comment by joshuahansel** [comment]
@GiudGiud

**Comment by GiudGiud** [comment]
This would be cleaner than erroring on `missing T_from_v_e`

which is problematic because we (my fault mostly) have been adding many of these T_from_v_e, as long as it was analytically tractable to do so

**Comment by joshuahansel** [comment]
This also suggests that maybe there's a problem with the design of `SinglePhaseFluidProperties`. Maybe it should be something like
```
SinglePhaseFluidPropertiesBase->CompressibleSinglePhaseFluidProperties
SinglePhaseFluidPropertiesBase->IncompressibleSinglePhaseFluidProperties
```

**Comment by GiudGiud** [comment]
could be
SinglePhaseFluidPropertiesBase->CompressibleSinglePhaseFluidProperties->IncompressibleSinglePhaseFluidProperties

I think the compressible ones work in incompressible solvers as long as the mach number is not too high

**Comment by joshuahansel** [comment]
Incompressible flow models always confuse me. So there, "incompressible" means more than actually being incompressible - it means constant density, right? But it doesn't require the equation of state to have a constant density?

**Comment by lindsayad** [comment]
I do agree that incompressible is a sub-class of compressible

**Comment by joshuahansel** [comment]
After more thought, I think the current design plus the method in the description is the right way to go.

**Comment by joshuahansel** [comment]
The ideal solution is something like:
```
FluidBlah : public FlowModelACompatible, public FlowModelBCompatible
```
where `FlowModelACompatible` would have pure virtual methods for all of the interfaces needed by the flow model A. I don't know off the top of my head the rules of multiple inheritance of virtual methods, since there's overlapping interface needs. If the overlapping works fine, it'd be a nice way to enforce compatibility.

**Comment by GiudGiud** [comment]
There's definitely overlapping interface needs. 
diamond inheritance isnt well supported

**Comment by joshuahansel** [comment]
Yeah, I thought as much. In any case, what we have now is fine. Just need to add `isCompressible()`.
